jib-core/src/main/java/com/google/cloud/tools/jib/http/BlobHttpContent.java

```diff
@@ -54,14 +58,10 @@ public boolean retrySupported() {
 
   @Override
   public void writeTo(OutputStream outputStream) throws IOException {
-    outputStream =
-        blobProgressListener == null
-            ? outputStream
-            : new ListenableCountingOutputStream(
-                outputStream,
-                blobProgressListener::handleByteCount,
-                blobProgressListener.getDelayBetweenCallbacks());
-    blob.writeTo(outputStream);
-    outputStream.flush();
+    try (DelayedConsumer<Long> delayedCountListener =
```

Comment:
This use of the DelayedConsumer seems like a decision that should be made by the creator of this class.

Reply:
That's a good point.